### PR TITLE
Solves SSH Key problem when admin user already exists

### LINKED ISSUE #2445: Web SSH Key not added when key is already attached to server
As both the debian package and the web ui use the `admin` user, the web ui will fail to add the ssh key, resulting in some users not having access to dokku push.

We should instead use `admin-web` as the admin user from the web ui if we detect that there is an existing `admin` user.

[Here](https://github.com/dokku/dokku/blob/master/contrib/dokku-installer.py#L103) is the line of code that executes this. What I'd do is list users, check if `admin` is in the list, and if so, change the username there from `admin` to `admin-web`.

### PR DESCRIPTION
**Pull request by mbtamuli**
This PR resolves #2445 

Suggestions are welcome if further changes are required.

### CODE REVIEW

File: contrib/dokku-installer.py [hunk 1]
```diff
@@ -119,6 +123,16 @@ def do_POST(self):
         self.end_headers()
         self.wfile.write(json.dumps({'status': 'ok'}))
 
+    def admin_user_exists(self):
+        command = 'dokku ssh-keys:list'
+        pattern = re.compile(r'NAME="admin"')
+        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+        for line in proc.stdout:
+            if pattern.search(line):
```
Comment (josegonzalez) [inline contrib/dokku-installer.py]: You can just do `return pattern.search(line)`

File: contrib/dokku-installer.py [hunk 2]
```diff
@@ -100,7 +101,10 @@ def do_POST(self):
...
             f.write(params['hostname'].value)
 
-        command = ['sshcommand', 'acl-add', 'dokku', 'admin']
+        user = 'admin'
+        if self.admin_user_exists():
+            user = 'web-admin'
+        command = ['sshcommand', 'acl-add', 'dokku', user]
         for key in params['keys'].value.split("\n"):
             proc = subprocess.Popen(command, stdin=subprocess.PIPE)
             proc.stdin.write(key)
```
Comment (josegonzalez) [inline contrib/dokku-installer.py]: Each key will need a new username...

Comment (josegonzalez) [inline contrib/dokku-installer.py]: Right but the web ui allows splitting the keys via newline, and as new sshcommand doesn't allow duplicate usernames, we'll need to modify the user for each successive write.

Comment (mbtamuli) [inline contrib/dokku-installer.py]: Okay, so the usernames should be asked though the form(web UI) or the usernames should be generated? Like `web-admin1`, `web-admin2`, etc?

Comment (josegonzalez) [inline contrib/dokku-installer.py]: Yep, that works for me :)

File: contrib/dokku-installer.py [hunk 3]
```diff
@@ -123,15 +132,28 @@ def do_POST(self):
...
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
-        for line in proc.stdout:
-            if pattern.search(line):
-                return True
-            else:
-                return False
+    return self.user_exists('admin(\d+)')
+
+    def user_exists(self, name):
+    command = 'dokku ssh-keys:list'
```
Comment (josegonzalez) [inline contrib/dokku-installer.py]: This isn't properly indented.

File: contrib/dokku-installer.py [hunk 4]
```diff
@@ -123,15 +132,28 @@ def do_POST(self):
         self.end_headers()
         self.wfile.write(json.dumps({'status': 'ok'}))
 
+    def web_admin_user_exists(self):
+    return self.user_exists('web-admin(\d+)')
```
Comment (josegonzalez) [inline contrib/dokku-installer.py]: Not properly indented.

**Pull request merged**